OnPod/Node/RearNode/main.cpp

```diff
@@ -0,0 +1,56 @@
+#include <Arduino.h>
+#include <pb_encode.h>
+#include <pb_decode.h>
+#include <SPI.h>
+
+#include "Sodaq_wdt.h"
+#include "Timer.h"
+
+#include "NodeStructs.h"
+#include "Paradigm.pb.h"
+
+#include "drivers/node_ethernet.h"
+#include "drivers/adc.h"
+#include "sensors/thermocouple.h"
+
+const NodeType NODE_TYPE = REAR;
+
+// timer
+Timer txTimer;
+const uint32_t TX_INTERVAL_MS = 50;
+
+// sensors
+SPIClass adcSPI (&PERIPH_SPI1, MISO1, SCK1, MOSI1, PAD_SPI1_TX, PAD_SPI1_RX);
+ADS7953 adc(adcSPI);
+typeKThermo thermo(&adc, 0);
+
+// comms
+UDPParams IPPara;  // keep IPs and ports in one location
+UDPClass udp(PIN_SPI_SS, IPAddress(IPPara.REAR_NODE_IP), IPPara.NODE_RX_PORT, NODE_TYPE);
+RearNodeToFc pRearNodeTelemetry = RearNodeToFc_init_default;  // protobuf message
+
+void sendToFlightComputer(void*) {
+    // create an output stream that writes to the UDP buffer
+    pb_ostream_t outStream = pb_ostream_from_buffer(udp.uSendBuffer, sizeof(udp.uSendBuffer));
+    // encode the message object and store it in the UDP buffer
+    pb_encode(&outStream, RearNodeToFc_fields, &pRearNodeTelemetry);
+    udp.sendPacket(IPAddress(IPPara.BBB_IP), IPPara.BBB_PORT, outStream.bytes_written);
+}
+
+void setup() {
+    udp.init();
+    adc.init();
```

Comment:
might be a dumb question but where are we enabling the adc's or just all on?

Reply:
They get enabled in the `init()` methods of each sensor